internal/querycoordv2/handlers.go

```diff
@@ -147,7 +148,32 @@ func (s *Server) balanceSegments(ctx context.Context, req *querypb.LoadBalanceRe
 		}
 		tasks = append(tasks, task)
 	}
-	return task.Wait(ctx, Params.QueryCoordCfg.SegmentTaskTimeout, tasks...)
+	return waitBalanceTasks(ctx, Params.QueryCoordCfg.SegmentTaskTimeout, tasks...)
+}
+
+// waitBalanceTasks blocks until the tasks finished,
+// return the first returned err of task,
+// NOTE: this ignores the TaskStale error,
+// because the balancing segments may be compacted,
+// which should be ignored, more details #19386
+func waitBalanceTasks(ctx context.Context, timeout time.Duration, tasks ...task.Task) error {
```

Comment:
Maybe `waitBalanceTasks` shall be in `task/utils.go` with `Wait` function.
Or we could change the `task.Wait` function add error check method?